js/slideshow.js

```diff
@@ -187,6 +194,79 @@
 		},
 
 		/**
+		 * Calculates the luminance of an image
+		 *
+		 * @param {*} image
+		 * @param {string} mimeType
+		 *
+		 * @returns {string}
+		 * @private
+		 */
+		_calculateBackgroundColour: function (image, mimeType) {
```

Comment:
This function could be split in two methods: `_calculateBackgroundColour(image, mimeType)` and `_isMainlyDark(image)`:

```js
_calculateBackgroundColour: function (image, mimeType) {
    var backgroundColour = this.darkBackgroundColour;
    if (!this._isTransparent(mimeType)) { // image.mimeType seems to be a typo in your original code
        return this.darkBackgroundColour;
    }
    if (this._isMainlyDark(image)) {
        return this.lightBackgroundColour;
    } else {
        return this.darkBackgroundColour;
    }
```